public/static-checkout/generate_static_pages.py

```diff
@@ -15,9 +15,9 @@
 
 # Read in course data and generate a page for each course
 with open('courselist.csv', newline='') as csvfile:
-    coursehtml = (html + '.')[:-1]
     csvreader = csv.reader(csvfile, delimiter=',')
     for row in csvreader:
+        coursehtml = (html + '.')[:-1]
```

Comment:
the deepcopy of the html  was happening outside the loop, so the html from the first course was being reused

Reply:
Great catch!